### PR TITLE
No need to add an instance in `WebApiCompatShimMvcBuilderExtensions`

### PR DESCRIPTION
**Pull request by dougbu**
- DI has changed since this class was written

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.WebApiCompatShim/WebApiCompatShimMvcBuilderExtensions.cs [hunk 1]
```diff
@@ -18,9 +18,7 @@ public static IMvcBuilder AddWebApiConventions(this IMvcBuilder builder)
             builder.Services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IConfigureOptions<WebApiCompatShimOptions>, WebApiCompatShimOptionsSetup>());
 
-            // The constructors on DefaultContentNegotiator aren't DI friendly, so just
-            // new it up.
-            builder.Services.TryAdd(ServiceDescriptor.Instance<IContentNegotiator>(new DefaultContentNegotiator()));
+            builder.Services.TryAddSingleton<IContentNegotiator, DefaultContentNegotiator>();
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.WebApiCompatShim/WebApiCompatShimMvcBuilderExtensions.cs]: Could just as easily be transient. But that would only create more `DefaultContentNegotiator` instances without changing behaviour (that I've seen). `DefaultContentNegotiator` is not stateful.

**Comment by rynowak** [comment]
:shipit:

**Comment by dougbu** [comment]
29a25b2

**Pull request closed without merging**